src/tiled/createobjecttool.h

```diff
@@ -66,6 +69,8 @@ class CreateObjectTool : public AbstractObjectTool
     ObjectGroupItem *mObjectGroupItem;
     MapObjectItem *mNewMapObjectItem;
     MapObjectItem *mOverlayPolygonItem;
+    bool mExtending;
+    bool mExtendingFirst;
```

Comment:
That makes no sense, so I quickly searched where these functions are being called.

Turns out it's because the `CreatePolylineObjectTool` is explicitly calling the functions of the `CreateObjectTool` class here:
```c++
void CreatePolylineObjectTool::finishNewMapObject()
{
    if (mNewMapObjectItem->mapObject()->polygon().size() >= 2)
        CreateObjectTool::finishNewMapObject();
    else
        CreateObjectTool::cancelNewMapObject();
}
```
I see no reason why it should be doing that. There are no existing overrides that would need to be skipped. So please just remove the `CreateObjectTool::` part there.